src/middlewares/log/index.ts

```diff
@@ -0,0 +1,87 @@
+import { Context } from 'hono';
+
+let logId = 0;
+
+// Map to store all connected log clients
+const logClients: Map<string | number, any> = new Map();
+
+const addLogClient = (clientId: any, client: any) => {
+  logClients.set(clientId, client);
+  console.log(
+    `New client ${clientId} connected. Total clients: ${logClients.size}`
+  );
+};
+
+const removeLogClient = (clientId: any) => {
+  logClients.delete(clientId);
+  console.log(
+    `Client ${clientId} disconnected. Total clients: ${logClients.size}`
+  );
+};
+
+const broadcastLog = async (log: any) => {
+  const message = {
+    data: log,
+    event: 'log',
+    id: String(logId++),
+  };
+
+  const deadClients: any = [];
+
+  // Run all sends in parallel
+  await Promise.all(
+    Array.from(logClients.entries()).map(async ([id, client]) => {
+      try {
+        await Promise.race([
+          client.sendLog(message),
+          new Promise((_, reject) =>
+            setTimeout(() => reject(new Error('Send timeout')), 1000)
+          ),
+        ]);
+      } catch (error: any) {
+        console.error(`Failed to send log to client ${id}:`, error.message);
+        deadClients.push(id);
+      }
+    })
+  );
+
+  // Remove dead clients after iteration
+  deadClients.forEach((id: any) => {
+    removeLogClient(id);
+  });
+};
+
+export const logger = () => {
+  return async (c: Context, next: any) => {
+    c.set('addLogClient', addLogClient);
+    c.set('removeLogClient', removeLogClient);
+
+    const start = Date.now();
+
+    await next();
+
+    const ms = Date.now() - start;
+    if (!c.req.url.includes('/v1/')) return;
+
+    const requestOptionsArray = c.get('requestOptions');
+    if (requestOptionsArray[0].requestParams.stream) {
+      requestOptionsArray[0].response = {
+        message: 'The response was a stream.',
+      };
+    } else {
+      const response = await c.res.clone().json();
```

Comment:
I think this will add latency to the actual response because the client will not receive the response until this whole code gets executed. Worth checking before merging this.

Reply:
Yes, the clone and broadcastLog will be executed along with the request. I'm hoping this only happens when someone runs this locally. Is there a way to move it to async?

Reply:
> I'm hoping this only happens when someone runs this locally

There is no check for this part in code. So it will get executed for all the requests.

Reply:
The logger middleware is attached only in start-server.ts and when there's no headless mode Param passed.

Reply:
For `getRuntimeKey() === "workerd"`, we can use `c.executionCtx.waitUntil()`

And for `getRuntimeKey() === "node"`, we can use `.then().catch()`. This one needs to be verified once.